test/integration/scheduler/framework_test.go

```diff
@@ -1284,3 +1290,92 @@ func TestCoSchedulingWithPermitPlugin(t *testing.T) {
 		cleanupPods(cs, t, []*v1.Pod{waitingPod, signallingPod})
 	}
 }
+
+// TestPreemptWithPermitPlugin tests preempt with permit plugins.
+func TestPreemptWithPermitPlugin(t *testing.T) {
```

Comment:
It tests the case: preempt works with permit plugins. If we want to verify that the permit plugin has received the rejected message from `scheduler.pereempt` and returned, we need to add a function for `WaitingPod` to retrieve the status and message, then we could verify it is from `scheduler.pereempt`. It seems not worth to add it just for this reason, so I just write a this test to test the preempt with permit plugins case.